File: sht3898/swtest/programmers/K번째수.py
# 내 풀이
# def solution(array, commands):
#     answer = []
#     for i in commands:
#         arr=sorted(array[i[0]-1:i[1]])
#         answer.append(arr[i[2]-1])
#     return answer


# 다른 사람 풀이
def solution(array, commands):
    return list(map(lambda command: sorted(array[command[0]-1:command[1]])[command[2]-1], commands))

File: Programmers/level1/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for(int c=0; c<commands.length; c++){
            ArrayList<Integer> arr = new ArrayList<Integer>();

            for(int idx=commands[c][0]-1; idx<commands[c][1]; idx++){
                arr.add(array[idx]);
            }
            Collections.sort(arr);
            answer[c] = arr.get(commands[c][2]-1);
        }
        return answer;
    }
}
